contracts/NFTAuction.sol

```diff
@@ -16,14 +16,14 @@ contract NFTAuction {
     //Each Auction is unique to each NFT (contract + id pairing).
     struct Auction {
         //map token ID to
-        uint256 minPrice;
-        uint256 buyNowPrice;
-        uint256 auctionBidPeriod; //Increments the length of time the auction is open in which a new bid can be made after each bid.
-        uint256 auctionEnd;
-        uint256 nftHighestBid;
-        uint256 bidIncreasePercentage;
-        uint256[] batchTokenIds; // The first token in the batch is used to identify the auction (contract + id pairing).
+        uint32 bidIncreasePercentage;
         uint32[] feePercentages;
+        uint128 minPrice;
```

Comment:
Dope!! Maybe the uint 32 can go further down and get packed together with an address?

Did you find out how the arrays were handled? I think they take a full 256 slot I am guessing?